src/lib/storage/run_length_segment/run_length_segment_iterable.hpp

```diff
@@ -44,143 +43,189 @@ class RunLengthSegmentIterable : public PointAccessibleSegmentIterable<RunLength
  private:
   const RunLengthSegment<T>& _segment;
 
+  /**
+   * Due to the nature of Run Length encoding, point-access is not in O(1).
+   * Run length segments store the end positions of runs in a sorted vector. To access a particular position, this
+   * vector has to be searched for the end position of the according run. It is possible to find the value of a
+   * position in O(log(n)) by doing a binary search.
+   * In case we hav already searched for a nearby position (e.g., when a previous not selective filter removed only
+   * relatively few lines), we can linearly search the end position. Since position lists are often ordered and modern
+   * CPUs are well at prefetching, linear searches are often faster than a complete binary search.
+   * determine_linear_search_offset_distance_threshold() estimates the threshold of when to use a linear or a binary
+   * search. Given the previous and the current chunk offset, the threshold is used to determine if a linear search is
+   * faster for the given number of skipped positions.
+   */
+
+  // LINEAR_SEARCH_ELEMENTS_THRESHOLD denotes the size of the range for that a linear search is faster than using a
+  // binary search on a sorted vector. The value of 200 has been found by a set of simple TPC-H measurements (#2038).
+  static constexpr auto LINEAR_SEARCH_VECTOR_DISTANCE_THRESHOLD = 200.0f;
+
+  // This methods estimates the threshold up to which is linear search is faster. Since the end position vector is
```

Comment:
But above you have defined the threshold to be 200. For me, the naming is a bit unclear.

Reply:
Updated the comment. It's about the threshold for the given RLE-end positions vector.

Does that make it clear?